packages/dataorc-utils/src/dataorc_utils/config/manager.py

```diff
@@ -0,0 +1,159 @@
+"""
+Parameter management for pipeline configuration.
+
+This manager reads configuration from environment variables.
+Cluster environment variables / wheel packaging.
+"""
+
+import os
+from enum import Enum
+from typing import Dict
+
+from .enums import CoreParam, Defaults, Environment
+from .models import CorePipelineConfig, InfraContext
+
+
+class PipelineParameterManager:
+    """
+    General parameter manager for data pipelines.
+
+    This manager is designed to work with repository-specific configurations.
+    Most repositories should create their own wrapper that provides the
+    repository-specific configuration dictionaries.
+    """
+
+    def __init__(
+        self,
+        environments_config: dict = None,
```

Comment:
What's the idea behind this `environments_config`?

Its name (and the documentation you've written) suggests that it can be used to override configuration per environment, but why would we care about the config for the test and prod environment when we are running a pipeline in the context of the dev environment? :thinking: